libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -98,7 +100,15 @@ internal class MapboxTripSession(
             is SetRefreshedRoutesInfo -> {
                 if (routes.isNotEmpty()) {
                     val primaryRoute = routes.first()
-                    navigator.refreshRoute(primaryRoute).fold(
+                    var refreshRouteResult: Expected<String, List<RouteAlternative>>? = null
+                    for (route in routes) {
+                        refreshRouteResult = navigator.refreshRoute(route)
+                    }
+                    if (refreshRouteResult!!.isValue) {
+                        this@MapboxTripSession.primaryRoute = routes.first()
+                        roadObjects = primaryRoute.upcomingRoadObjects
+                    }
```

Comment:
Why do we need this block? Same happens in the `fold` function below.

Reply:
Right. Rebase issues.